From 9f974046c37b7c28705d5558328759fff708b1cb Mon Sep 17 00:00:00 2001
From: Yaakov Selkowitz <yselkowi@redhat.com>
Date: Fri, 27 Jun 2025 04:54:08 -0500
Subject: Fix build with glibc-2.42
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

exportfs.c: In function ‘release_lockfile’:
exportfs.c:83:17: error: ignoring return value of ‘lockf’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
   83 |                 lockf(_lockfd, F_ULOCK, 0);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
exportfs.c: In function ‘grab_lockfile’:
exportfs.c:77:17: error: ignoring return value of ‘lockf’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
   77 |                 lockf(_lockfd, F_LOCK, 0);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~

lockf is now marked with attribute warn_unused_result:

https://sourceware.org/git/?p=glibc.git;a=commitdiff;h=f3c82fc1b41261f582f5f9fa12f74af9bcbc88f9

Signed-off-by: Steve Dickson <steved@redhat.com>

Upstream: https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commit;h=9f974046c37b7c28705d5558328759fff708b1cb

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 utils/exportfs/exportfs.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/utils/exportfs/exportfs.c b/utils/exportfs/exportfs.c
index b03a047b..748c38e3 100644
--- a/utils/exportfs/exportfs.c
+++ b/utils/exportfs/exportfs.c
@@ -74,13 +74,19 @@ grab_lockfile(void)
 {
 	_lockfd = open(lockfile, O_CREAT|O_RDWR, 0666);
 	if (_lockfd != -1)
-		lockf(_lockfd, F_LOCK, 0);
+		if (lockf(_lockfd, F_LOCK, 0) != 0) {
+			xlog_warn("%s: lockf() failed: errno %d (%s)",
+			__func__, errno, strerror(errno));
+		}
 }
 static void
 release_lockfile(void)
 {
 	if (_lockfd != -1) {
-		lockf(_lockfd, F_ULOCK, 0);
+		if (lockf(_lockfd, F_ULOCK, 0) != 0) {
+			xlog_warn("%s: lockf() failed: errno %d (%s)",
+			__func__, errno, strerror(errno));
+		}
 		close(_lockfd);
 		_lockfd = -1;
 	}
-- 
cgit v1.2.3

